vec-common/src/main/java/com/foursoft/vecmodel/common/HasSpecifications.java

```diff
@@ -28,12 +28,27 @@
 import com.foursoft.vecmodel.common.util.DelegationUtils;
 
 import java.util.List;
+import java.util.Optional;
 
-public interface HasSpecifications<X> {
+public interface HasSpecifications<X extends HasIdentification> {

```

Comment:
Those `HasX` classes could all be marked as a `@FunctionalInterface`.

Reply:
Good idea, you can do this in new branch.

Reply:
I don't see an issue not to do it in this branch. Why would you want a new PR for that?